### ISSUE TITLE
gvim gtk - "dirt" on left border when glyph larger than caret is drawn

### ISSUE THREAD

**Issue by salty-horse** [vim/vim#4130]
Using vim 8.1 with patches 1-1004.

Steps to reproduce:
1) Run `gvim --clean`
2) `:set guifont=Ubuntu\ Mono\ 12` (You may need to [download it](https://design.ubuntu.com/font/).)
3) Enter the character `▸` at the start of a line.
4) Delete the character.

Result:
A column of pixels will be visible at the left border where the left-most column of the deleted character was, until the window is resized.

See screenshots:
![character_large](https://user-images.githubusercontent.com/28355/54604110-a81eb500-4a4e-11e9-867f-6561844bc0a9.png)
![deleted_large](https://user-images.githubusercontent.com/28355/54604131-b371e080-4a4e-11e9-8c92-711a48b0b093.png)

**Comment by chrisbra** [comment]
Does it make a difference if you use a different size? What about other fonts?

**Comment by salty-horse** [comment]
The font size doesn't matter, because gvim resizes its dimensions accordingly.

I assume it can also happen with other fonts, depending on how wide their glyphs are. I specified this font because the default font gvim uses on my system has a smaller `▸` that doesn't exhibit this problem.

**Comment by lilydjwg** [comment]
I sometimes see this issue too. Ctrl-L will clear them.

**Comment by chrisbra** [comment]
Yes, but I wonder whether this is a problem of the font of or of gvim.

**Comment by tonymec** [comment]
I sometimes had this problem with fonts which pretended to be monotype but weren't — for instance, IIRC, the Lucida Typewriter glyphs for bold Cyrillic were obtained by superimposing two copies of the corresponding unbold glyphs, offset by one pixel horizontally, with the result that they were actually "one pixel wider" than advertised. Setting a different font family in 'guifont' made the problem disappear.

**Comment by salty-horse** [comment]
It's hard for fonts to be fully monotype - I want and expect fallbacks to other fonts when a glyph isn't available, and it's ok when those fonts are not monotype - `𒈙` (U+12219) is handled just fine.

Is this a user error? Gvim "allows" it to be drawn on parts of the window that it doesn't expect - I don't think the user is at fault for picking a "wrong" font.

**Comment by chrisbra** [comment]
no it is not a user error, but one can expect some visual glitches when a certain glyph does not fully fit into the cell.
